crates/core_simd/src/fmt.rs

```diff
@@ -1,39 +1,20 @@
 use crate::simd::{LaneCount, Simd, SimdElement, SupportedLaneCount};
 use core::fmt;
 
-macro_rules! impl_fmt_trait {
-    { $($trait:ident,)* } => {
-        $(
-            impl<T, const LANES: usize> fmt::$trait for Simd<T, LANES>
-            where
-                LaneCount<LANES>: SupportedLaneCount,
-                T: SimdElement + fmt::$trait,
-            {
-                fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-                    #[repr(transparent)]
-                    struct Wrapper<'a, T: fmt::$trait>(&'a T);
-
-                    impl<T: fmt::$trait> fmt::Debug for Wrapper<'_, T> {
-                        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-                            self.0.fmt(f)
-                        }
-                    }
-
-                    f.debug_list()
-                        .entries(self.as_array().iter().map(|x| Wrapper(x)))
-                        .finish()
-                }
-            }
-        )*
+impl<T, const LANES: usize> fmt::Debug for Simd<T, LANES>
+where
+    LaneCount<LANES>: SupportedLaneCount,
+    T: SimdElement + fmt::Debug,
+{
+    /// A `Simd<T, N>` has a debug format like the one for `[T]`:
+    /// ```
+    /// # #![feature(portable_simd)]
+    /// # #[cfg(feature = "as_crate")] use core_simd::simd::Simd;
+    /// # #[cfg(not(feature = "as_crate"))] use core::simd::Simd;
+    /// let floats = Simd::<f32, 4>::splat(-1.0);
+    /// assert_eq!(format!("{:?}", [-1.0; 4]), format!("{:?}", floats));
+    /// ```
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
```

Comment:
Since this is just passed to array's implementation, I think this function should be inline

Reply:
@bjorn3 @Sp00ph What's your estimation of the binary size impact on sth like a `libcore.so` of doing that, here?

Reply:
Actually this PR doesn't change binary size of `libcore.o` at all, because it's `core::core_arch::simd::` that's using space, not portable simd 😅

Reply:
(But I still think it's a good cleanup and it's good to match array in terms of formatting)

Reply:
> Actually this PR doesn't change binary size of `libcore.o` at all, because it's `core::core_arch::simd::` that's using space, not portable simd 😅

`core::arch::simd` uses `portable_simd` all over the place internally though.

> @Sp00ph What's your estimation of the binary size impact on sth like a `libcore.so` of doing that, here?

In my testing the .rlib shrunk by about 65K. Not sure how well that translates to the size of the `.so` though.

Reply:
> Actually this PR doesn't change binary size of libcore.o at all, because it's `core::core_arch::simd::` that's using space, not portable simd :sweat_smile:

oh right, lemme see...

> core::arch::simd uses portable_simd all over the place internally though.

Eh not quite.

Reply:
`core::core_arch::simd` is from `stdarch` crate, not `portable-simd`.

When I say "doesn't change binary size at all", it really means it! I looked at the `size core.o` and the output is exactly the same.

Reply:
My argument for `inline` has less to do with binary size and more that this function simply defers to another function, so it's expected that this will inline (whether or not slice fmt is marked `inline` is another story)

Reply:
Hm. Are you accounting for the bottom-up nature of inlining, and thus at the time this function has an inlining decision made on it, it may already contain all the fmt code instantiated inside it, thus a hint to strengthen the inlining would reduce the likelihood of what we want (a single fmt instance)?

Reply:
I know it's not quite this simple, but if the slice fmt function is inlined into this function, it should probably be inlined into the caller.  If it isn't, I suspect this function will be inlined regardless of what attribute we put on it.

Reply:
Fair enough!